src/backend/pluginPlan.test.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+
+import {Graph} from "../core/graph";
+import {node as graphNode} from "../core/graphTestUtil";
+import {createProject} from "../core/project";
+import {TestTaskReporter} from "../util/taskReporter";
+import {pluginMirrorPlan} from "./pluginPlan";
+
+const githubSentinel = graphNode("github-sentinel");
+const githubGraph = () => new Graph().addNode(githubSentinel);
+const discourseSentinel = graphNode("discourse-sentinel");
+const discourseGraph = () => new Graph().addNode(discourseSentinel);
+const combinedGraph = () => Graph.merge([githubGraph(), discourseGraph()]);
+
+const mockCacheProvider = () => ({
+  database: jest.fn(),
+});
+
+const mockPluginLoaders = () => ({
+  github: {
+    updateMirror: jest.fn(),
+    createGraph: jest.fn().mockResolvedValue(githubGraph()),
+  },
+  discourse: {
+    updateMirror: jest.fn(),
+    createGraph: jest.fn().mockResolvedValue(discourseGraph()),
+  },
+  identity: {
+    contractGraph: jest.fn().mockResolvedValue(combinedGraph()),
+  },
+});
+
+describe("src/backend/pluginPlan", () => {
+  describe("pluginMirrorPlan", () => {
+    it("should update discourse mirror", async () => {
+      // Given
+      const loaders = mockPluginLoaders();
+      const cache = mockCacheProvider();
+      const reporter = new TestTaskReporter();
+      const ghToken = null;
+      const project = createProject({
+        id: "has-discourse",
+        discourseServer: {serverUrl: "http://foo.bar"},
+      });
+
+      // When
+      const mirror = pluginMirrorPlan(loaders, ghToken, cache, reporter);
+      await mirror(project);
+
+      // Then
+      const {discourse, github, identity} = loaders;
```

Comment:
This test is currently comprehensive in that it verifies none of the other loaders were called inappropriately. However, it strikes me as quite unlikely that the next person to add a loader will update this particular test case, at which point the test is verifying that some (old) subset of the loaders are not being called inappropriately.

I think we can make things a bit simpler by having the test only verify that the discourse mirror was updated correctly, without verifying that other methods weren't called.